wlog-zap/internal/marshalers/wtracing_spanmodel.go

```diff
@@ -49,10 +49,30 @@ func marshalWTracingSpanModel(key string, val interface{}) zapcore.Field {
 				}
 			}
 		}
+		if tags := span.Tags; tags != nil && len(tags) > 0 {
+			enc.AddObject("tags", newTagEncoder(tags))
+		}
 		return nil
 	}))
 }
 
+type tagEncoder struct {
```

Comment:
Using the non-exported tagEncoder seemed better than using `enc.Reflect`. I believe the creation of the object will only result in one pointer being created.

Reply:
This looks fine by me, but I think you can use some convenience methods provided by zap, similar to how it's done in `encodeSpanModelAnnotations` below.
For example, zap has a `zapcore.ObjectMarshalerFunc()` which let's you inline all of the logic for marshaling the object.
```go
func encodeSpanTags(enc zapcore.ObjectEncoder, tags map[string]string) error {
	return enc.AddObject(trc1log.SpanTagsKey, zapcore.ObjectMarshalerFunc(func(objEnc zapcore.ObjectEncoder) error {
		for k, v := range tags {
			objEnc.AddString(k, v)
		}
		return nil
	}))
}

```

Reply:
That's what I wanted! Thanks.